pkg/apis/core/validation/cloudprofile.go

```diff
@@ -211,66 +167,21 @@ func validateMachineTypeStorage(storage core.MachineTypeStorage, fldPath *field.
 	return allErrs
 }
 
-func validateMachineImages(machineImages []core.MachineImage, fldPath *field.Path) field.ErrorList {
+func validateMachineImagesConfiguration(machineImages []core.MachineImage, fldPath *field.Path) field.ErrorList {
```

Comment:
```suggestion
func validateCPImages(machineImages []core.MachineImage, fldPath *field.Path) field.ErrorList {
```

Nit: To separate CP validations from NCP (devs usually don't see them in IDEs etc. because they are in the same package).

Reply:
I've also refactored some functions again and moved commonly used ones to `utils.go` with a neutral name.